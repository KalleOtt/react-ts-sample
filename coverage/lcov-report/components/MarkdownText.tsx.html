<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/MarkdownText.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\prettify.css" />
    <link rel="stylesheet" href="..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\index.html">All files</a> / <a href="index.html">components</a> MarkdownText.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.87% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>26/31</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">55.17% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>16/29</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>7/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as marked from 'marked'<span class="cstat-no" title="statement not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="missing-if-branch" title="else path not taken" >E</span><span class="missing-if-branch" title="else path not taken" >E</span></span>
import * as React from 'react'<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="branch-1 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="else path not taken" >E</span><span class="branch-1 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="branch-2 cbranch-no" title="branch not covered" ><span class="branch-1 cbranch-no" title="branch not covered" ><span class="branch-0 cbranch-no" title="branch not covered" ><span class="missing-if-branch" title="if path not taken" >I</span><span class="missing-if-branch" title="if path not taken" >I</span></span></span></span></span></span></span></span></span>
import { Component, CSSProperties } from 'react'
&nbsp;
interface MarkdownTextProps {
    readonly text: string
    style?: CSSProperties
}
&nbsp;
marked.setOptions({
    gfm: true,
    sanitize: false
})
&nbsp;
export class MarkdownText extends Component&lt;MarkdownTextProps, {}&gt; {
&nbsp;
    render() {
        const parsedMarkdown = marked.parse(this.pro<span class="missing-if-branch" title="else path not taken" >E</span>ps.text || '')<span class="cstat-no" title="statement not covered" ></span>
&nbsp;
        const mergedStyle: React.CSSProperties = Object.assign({
            color: 'black'
        }, this.props.style)
&nbsp;
        return (
            &lt;div style={mergedStyle} dangerouslySetInnerHTML={{ __html: parsedMarkdown }} /&gt;
        )
    }
}</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Mon Nov 21 2016 15:42:07 GMT+0100 (Mitteleurop√§ische Zeit)
</div>
</div>
<script src="..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\sorter.js"></script>
</body>
</html>
