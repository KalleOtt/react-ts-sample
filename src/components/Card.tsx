import glamorous from 'glamorous'
import { rgba } from 'rgba-string'

export const Card = glamorous.div(
  {
    background: '#fff',
    borderRadius: '2px',
    display: 'inline-block',
    height: '300px',
    margin: '1rem',
    position: 'relative',
    width: '300px',
  },
  ({ level = 1 }: { level?: number }) => {
    const first = level
    const shade1 = rgba(0, 0.12 + (level - 1) * 0.04)
    const shade2 = rgba(0, 0.24 - (level - 1) * Math.floor(level / 2))
    return {
      boxShadow: `0 ${first}px ${first ** first +
        level}px ${shade1}, 0 ${level ** level - (level - 1)}px ${level **
        level -
        (level - 1)}px ${shade2}`,
    }
  }
)
