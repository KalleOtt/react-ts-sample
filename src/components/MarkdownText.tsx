import * as marked from 'marked'
import * as React from 'react'
import { CSSProperties } from 'react'
import glamorous from 'glamorous'
import { rgba } from 'rgba-string'

type MarkdownTextProps = {
  readonly text: string
  readonly style?: CSSProperties
}

marked.setOptions({
  gfm: true,
  sanitize: false,
})

const TextDiv = glamorous.div({
  color: rgba(0x1e1e1e),
  fontFamily: 'Lato, sans-serif',
})

export const MarkdownText = (props: MarkdownTextProps) => {
  const parsedMarkdown = marked.parse(props.text || '')

  return (
    <TextDiv {...props} dangerouslySetInnerHTML={{ __html: parsedMarkdown }} />
  )
}
