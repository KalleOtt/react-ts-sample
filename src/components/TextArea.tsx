import * as React from 'react';
import { PureComponent, CSSProperties } from 'react';
import { Input } from 'react-toolbox/lib/input';

export interface TextAreaProps {
    onChange: (newValue: string) => void;
    disabled?: boolean;
    name?: string;
    value?: string;
    style?: React.CSSProperties;
    rows?: number;
}

interface TextAreaState {
    value: string;
}

export class TextArea extends PureComponent<TextAreaProps, TextAreaState> {

    state = {
        value: this.props.value || ''
    };

    componentWillReceiveProps = (nextProps: TextAreaProps) => {
        const { value } = nextProps;
        this.setState({ value: value || '' });
    }

    private changeHandler = (value: string) => {
        this.setState({ value });
        const { onChange } = this.props;
        onChange(value);
    }

    static style: CSSProperties = {
        maxHeight: '100%',
        height: '100%',
        overflow: 'auto',
        lineHeight: '20px'
    };

    render() {
        const { disabled, name, style, rows } = this.props;
        const { value } = this.state;
        const mergedStyle: CSSProperties = Object.assign({}, TextArea.style, style);

        return (
            <Input
                rows={rows}
                type={'text'}
                style={mergedStyle}
                multiline={true}
                disabled={disabled}
                value={value}
                name={name}
                onChange={this.changeHandler}
            />
        );
    }
}