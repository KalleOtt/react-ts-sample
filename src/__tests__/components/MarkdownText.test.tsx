import * as React from 'react';
import renderer from 'react-test-renderer';
import { MarkdownText } from 'components/MarkdownText';

describe('MarkdownText shall render text', () => {

    const textValue = '# headline\ntext';

    const component = renderer.create(
        <MarkdownText text={textValue} />
    );
    let tree = component.toJSON();

    test('Snapshot test', () => {
        expect(tree).toMatchSnapshot();
    });
});