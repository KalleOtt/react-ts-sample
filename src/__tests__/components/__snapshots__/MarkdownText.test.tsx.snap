// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownText shall render text Snapshot test 1`] = `
.glamor-0,
[data-glamor-0] {
  color: rgba(30, 30, 30, 1);
  font-family: Lato, sans-serif;
}

<div
  className="glamor-0"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<h1 id=\\"headline\\">headline</h1>
    <p>text</p>
    ",
    }
  }
/>
`;
